app.py

```diff
@@ -373,6 +373,50 @@ def _register_event_log(self, ctx, command_executed, version=None):
         desc =  "%s %s: %s" % (self.name, self.version, menu_name)
         tank.util.create_event_log_entry(self.tank, ctx, "Toolkit_App_Startup", desc, meta)
 
+    def prepare_generic_launch(self, engine_name, context, app_path, app_args):
+        """
+        Generic engine launcher.
+
+        This method will look for a bootstrap method in the engine's
+        python/startup/bootstrap.py file if it exists.  That bootstrap will be
+        called if possible.
+
+        :param engine_name: The name of the engine being launched
+        :param context: The context that the application is being launched in
+        :param app_path: Path to DCC executable or launch script
+        :param app_args: External app arguments
+
+        :returns: extra arguments to pass to launch
+        """
+        # find the path to the engine on disk where the startup script can be found:
+        engine_path = tank.platform.get_engine_path(engine_name, self.tank, context)
+        if engine_path is None:
+            raise TankError("Path to '%s' engine could not be found." % engine_name)
+
+        # find bootstrap file located in the engine and load that up
+        startup_path = os.path.join(engine_path, "python", "startup", "bootstrap.py")
+
+        if not os.path.exists(startup_path):
+            # Generic bootstrap is not supported, match the old behavior for an unsupported engine.
+            raise TankError("The %s engine is not supported!" % engine_name)
```

Comment:
This could also be more explanatory I think - so a user would hit this because the engine was found but the bootstrap file wasn't found. So this error is most likely to hit a *developer* rather than an artist. I would suggest something along the lines "Could not find bootstrap script  '/path/to/engine/python/startup/bootstrap.py' for the tk-modo engine!"